public/scripts/bookmarks.js

```diff
@@ -351,8 +370,314 @@ async function convertSoloToGroupChat() {
     toastr.success('The chat has been successfully converted!');
 }
 
-jQuery(function () {
+/**
+ * Creates a new branch from the message with the given ID
+ * @param {number} mesId Message ID
+ * @returns {Promise<string?>} Branch file name
+ */
+export async function branchChat(mesId) {
+    if (this_chid === undefined && !selected_group) {
+        toastr.info('No character selected.', 'Create Branch');
+        return null;
+    }
+
+    const fileName = await createBranch(mesId);
+    await saveItemizedPrompts(fileName);
+
+    if (selected_group) {
+        await openGroupChat(selected_group, fileName);
+    } else {
+        await openCharacterChat(fileName);
+    }
+
+    return fileName;
+}
+
+function registerBookmarksSlashCommands() {
+    /**
+     * Validates a message ID. (Is a number, exists as a message)
+     *
+     * @param {number} mesId - The message ID to validate.
+     * @param {string} context - The context of the slash command. Will be used as the title of any toasts.
+     * @returns {boolean} - Returns true if the message ID is valid, otherwise false.
+     */
+    function validateMessageId(mesId, context) {
+        if (isNaN(mesId)) {
+            toastr.warning('Invalid message ID was provided', context);
+            return false;
+        }
+        if (!chat[mesId]) {
+            toastr.warning(`Message for id ${mesId} not found`, context);
+            return false;
+        }
+        return true;
+    }
+
+    SlashCommandParser.addCommandObject(SlashCommand.fromProps({
+        name: 'branch-create',
+        returns: 'Name of the new branch',
+        callback: async (args, text) => {
+            const mesId = Number(args.mesId ?? text ?? getLastMessageId());
```

Comment:
I was looking at quite a few other commands, where we supported both unnamed arguments or named arguments for the same value. `/getvar` might be the most prominent example.
Though for a new slash command, there is no need to follow that principle, that is true...

If I make it unnamed, it's easier to type, sure. If I only make it named, we have a command with no unnamed arguments. Weird. But it's consistent, because the other commands here all take a `mes` argument for the ID, instead of unnamed argument.
Meeeeh.

Reply:
Removed all named if message is the only argument that can be provided.